### ISSUE TITLE
Adjust EPUB footnotes with backlink before text.

### ISSUE THREAD

**Issue by jgm** [jgm/pandoc#5583]
See #4491 which has been abandoned but has several good ideas in it, which I don't want to get lost.

[EPUB3 Accessibility Guidelines](http://kb.daisy.org/publishing/docs/html/notes.html) say

- Identify footnotes within the text using the aside element. [WCAG 1.3.1]
- Group footnotes or endnotes at the end of a section or work in lists. [WCAG 1.3.1]
- Link note references to their notes. [WCAG 1.3.1]
- Backlink notes at the end of a section or work to their references. [WCAG 1.3.1]
- For footnotes, the backlink precedes the note text so that the user will be aware that such a link is available.

All we really need to do is change the backlink to precede the text, and make sure all the links and cross-references and containing sections conform to guidelines.

See also:

- [Amazon kindle guidelines](https://kindlegen.s3.amazonaws.com/AmazonKindlePublishingGuidelines.pdf)
- [iBooks Asset Guide](https://help.apple.com/itc/booksassetguide/en.lproj/static.html#itccf8ecf5c8)
- [DPUB-Aria](https://www.w3.org/TR/dpub-aria-1.0/#doc-footnote)

**Comment by jgm** [comment]
Note however that under "backlink with endnote" it recommends putting the backlink at the end, so as not to interfere with note numbering.
```
<li role="doc-endnote">
   <p>According to the usual nomenclature, the …</p>
   <p><a href="#ref" role="doc-backlink">Go to note reference</a></p>
</li>
```

**Comment by lewer** [comment]
Currently, with epub3 writer, the `#footnotes` section has `epub:type="footnotes"` and every footnote has `<li epub:type="footnote">`

As I understand https://kb.daisy.org/publishing/docs/html/notes.html, "footnotes" are what is put in an aside block in the body, while "endnotes" are grouped at the end in a proper section.

So currently pandoc kind of mixes endnotes and foonotes: it uses endnotes but tags them with `epub:type="footnotes"`

To follow https://kb.daisy.org/publishing/docs/html/notes.html with minimal changes, what do you think of replacing `epub:type="footnotes"` with `epub:type="endnotes"` in the `#footnotes` section, add `role="doc-endnotes"` to this same section, and removing `epub:type="footnote"` on each individual footnote?

**Comment by jgm** [comment]
> @Porges concluded that backlink should be placed at end for footnotes, and at start for endnotes.

Isn't it the other way round?  Backlink should go at the end for endnotes, at the start for footnotes.
